src/main/java/org/codehaus/staxmate/dom/DOMConverter.java

```diff
@@ -167,21 +180,40 @@ public void buildDocument(XMLStreamReader r, Document doc)
          */
         boolean wholeDoc = (r.getEventType() == XMLStreamConstants.START_DOCUMENT);
 
+		nsmap=new java.util.HashMap();
+
         XMLStreamReader2 sr = Stax2ReaderAdapter.wrapIfNecessary(r);
         QNameRecycler recycler = new QNameRecycler();
         boolean nsAware = _isNamespaceAware(sr);
         Node current = doc; // At top level
+		int position=0;
```

Comment:
One quick note: indentation is off, maybe it's using tabs? If so, should convert to all spaces, makes diff easier to read.

Reply:
i have addressed this point in my latest commit